pettingzoo/classic/hanabi/hanabi.py

```diff
@@ -151,10 +151,14 @@ def __init__(self,
 
         # Set action_spaces and observation_spaces based on params in hanabi_env
         self.action_spaces = {name: spaces.Discrete(self.hanabi_env.num_moves()) for name in self.agents}
-        self.observation_spaces = {player_name: spaces.Box(low=0,
-                                                           high=1,
-                                                           shape=(self.hanabi_env.vectorized_observation_shape()[0],),
-                                                           dtype=np.float32)
+        self.observation_spaces = {player_name: spaces.Dict({'observation': spaces.Box(low=0,
+                                                                                       high=1,
+                                                                                       shape=(
+                                                                                       self.hanabi_env.vectorized_observation_shape()[
+                                                                                           0],),
+                                                                                       dtype=np.float32),
+                                                             'action_mask': spaces.Box(low=0, high=1, shape=(20,),
```

Comment:
The shape here can't be hard-coded. Like the action spaces, get the number of possible moves with `self.hanabi_env.num_moves()`